infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -171,33 +174,40 @@ public SequencerServer(ServerContext serverContext) {
                 && Long.parseLong((String) opts.get("--sequencer-cache-size")) != 0) {
             cacheSize = Long.parseLong((String) opts.get("--sequencer-cache-size"));
         }
+        if (opts.get("--sequencer-per-stream-cache-size") != null
+                && Long.parseLong((String) opts.get("--sequencer-per-stream-cache-size")) != 0) {
+            perStreamCacheSize =
+                    Long.parseLong((String) opts.get("--sequencer-per-stream-cache-size"));
+        }
+
 
         conflictToGlobalTailCache = Caffeine.newBuilder()
                 .maximumSize(cacheSize)
-                .removalListener((Integer k, Long v, RemovalCause cause) -> {
+                .removalListener((UUID k, Cache<Integer, Long> v, RemovalCause cause) -> {
                     if (!RemovalCause.REPLACED.equals(cause)) {
-                        log.trace("Updating maxConflictWildcard. Old value = '{}', new value='{}'"
-                                        + " conflictParam = '{}'. Removal cause = '{}'",
-                                maxConflictWildcard, v, k, cause);
-                        maxConflictWildcard = Math.max(v, maxConflictWildcard);
+                        if (v != null) {
```

Comment:
Well, the previous tests cover this code, but we could add some additional tests to make sure the wildcard is properly loaded on eviction.